### DISCUSSION TITLE
Why Is My OSD Not Working?

### DISCUSSION THREAD

**Discussion by Blake1532** [iNavFlight/inav#9666] [Support]
Hello,

I am setting up avatar fpv on my inav quad, and I discovered a very annoying issue.  The osd does not show up!  Let me explain what I've tried so far.  

- Direct Soldering instead of using plugs
- Switching uarts
- Using All msp modes including msp displayport
- Updating firmware on both the vtx and goggles
- Changing fonts

I have seen countless videos where people use avatar with inav.  So I think im just missing a cli line prompt or something else.  Can anyone help me here?

Thank you so much!
Blake

**Comment by MrD-RC** [comment]
You’re mentioning CLI. So it sounds like you’re trying to set this up like  BetaFlight. This is not the way. 

Setting up on INAV is very simple. On the ports, select **MSP DisplayPort** as the peripheral. Do not flip the MSP toggle! Then, save and reboot. Next, visit the OSD page and select Avatar from the dropdown list on the right. Everything is then set up. If it’s not working. It’s probably then a wiring issue. Do you have Tx and Rx reversed?

**Reply by SodaGremlin** [reply]
Specifically the part about not flipping the switch, and using the peripheral drop down was useful here! Thank you!

**Comment by Blake1532** [comment]
Hello

I have set it up exactly like that.  Turn on only MSP display port without
hitting the MSP slider, and select Avatar.  However it doesn't work for
some reason.

I thought that might be the case. However, I swapped the TX and RX wires
just to make sure that say the markings were wrong on the flight controller
or the VTX and that didn't do it either.

As far as I can tell. All of the Avatar OSD that I've seen on YouTube has
been done with a 6.xx version of the firmware, not the seven version, maybe
there's something wrong with it.  I think tomorrow I will downgrade to a
6.0.0 version and see if that solves my issues.

I have a theory that it also could be something to do with the walksnail
system itself. Maybe walksnail is trying to use some different font other
than inav is trying to use so it just doesn't show up.

In the past I followed Joshua bardwell's tutorial for how to upgrade the
fonts. I will also try factory resetting my goggles tomorrow to get rid of
any traces of alteration from the goggles.

Could there be something that I'm missing though? Should I maybe try to set
it to the alternate OSD profile number one and set it on a constantly
enabled switch? I don't know.

Thank you so much!
Blake.

**Reply by MrD-RC** [reply]
It works fine with 7.0. I’m using that on my aircraft. Of you were using the wrong font, you should see a jumbled OSD. 

Check the modes, to make sure you haven’t activated `OSD OFF`

**Reply by Blake1532** [reply]
Hello,

I have checked the modes, "OSD OFF" is not on.  Unfortunately, I don't know why.  It could be a dead OSD chip, in which case I'll ask for a refund because I haven't flown this aircraft yet.  Is there any way to know for sure that the OSD chip is dead?  Or could I send you guys a "DIFF ALL" so you can see if there is anything wrong with my configuration?

Thank you so much for all the help!
Blake

**Reply by Jetrell** [reply]
The digital systems don't use an OSD 7456 chip, as analogue does.. The OSD data is sent directly from the FC to the VTX, which superimposes the font onto the image. 
From what you have said. It appears you've set it up correctly in INAV.
Have you also looked at the Walksnail side..  Do you have the `userfont` folder loaded onto your SD card root directory ? Which contains the `.png` fonts and `.ini` update directory files.

**Reply by Blake1532** [reply]
Hello

Well, I did, but then to make sure, I factory re-setted the goggles to get rid of it, as well as wiped the sd card and did a full update.  I'm beginning to think the OSD chip is dead.  I saved my config from inav so I could go back.  I then uploaded betaflight, and I can't seem to get it working on betaflight either.

Thank you,
Blake

**Reply by Blake1532** [reply]
Hello,

Here's my diff all:

# version
# INAV/SPEEDYBEEF405V4 7.0.0 Dec  5 2023 / 11:06:00 (895a4f31)
# GCC-10.3.1 20210824 (release)

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# Timer overrides

# Outputs [servo]

# safehome

# features
feature GPS
feature PWM_OUTPUT_ENABLE

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS
blackbox -GYRO_RAW
blackbox -PEAKS_R
blackbox -PEAKS_P
blackbox -PEAKS_Y

# Receiver: Channel map
map TAER

# Ports
serial 0 0 115200 115200 0 115200
serial 1 8388608 115200 115200 0 115200
serial 2 64 115200 115200 0 115200
serial 3 0 115200 115200 0 115200
serial 4 0 115200 115200 0 115200
serial 5 2 115200 115200 0 115200

# LEDs

# LED color

# LED mode_color

# Modes [aux]
aux 0 0 3 1700 2100
aux 1 51 5 1700 2100
aux 2 1 6 1300 1650
aux 3 2 6 1750 2100
aux 4 13 0 1700 2100
aux 5 27 1 1700 2100

# Adjustments [adjrange]

# Receiver rxrange

# temp_sensor

# Mission Control Waypoints [wp]
#wp 0 invalid

# OSD [osd_layout]
osd_layout 0 0 49 11 V
osd_layout 0 1 1 1 V
osd_layout 0 5 46 1 V
osd_layout 0 6 46 3 V
osd_layout 0 7 23 12 V
osd_layout 0 8 19 1 V
osd_layout 0 9 24 18 V
osd_layout 0 11 1 7 V
osd_layout 0 12 1 5 V
osd_layout 0 13 48 9 V
osd_layout 0 14 48 13 V
osd_layout 0 15 47 7 V
osd_layout 0 20 42 18 V
osd_layout 0 21 1 18 V
osd_layout 0 22 25 10 V
osd_layout 0 23 47 5 V
osd_layout 0 28 23 11 H
osd_layout 0 30 11 13 V
osd_layout 0 32 1 3 V
osd_layout 0 35 1 9 V
osd_layout 0 46 1 13 V
osd_layout 0 108 23 11 V
osd_layout 0 138 1 11 V
osd_layout 0 142 28 1 V

# Programming: logic

# Programming: global variables

# Programming: PID controllers

# master
set looptime = 500
set gyro_main_lpf_hz = 90
set gyro_main_lpf_type = PT1
set dynamic_gyro_notch_q = 250
set dynamic_gyro_notch_min_hz = 60
set dynamic_gyro_notch_mode = 3D
set setpoint_kalman_q = 200
set gyro_zero_x = -19
set gyro_zero_y = -4
set ins_gravity_cmss =  977.851
set acc_hardware = ICM42605
set acczero_x = -3
set acczero_y = -6
set acczero_z = 14
set accgain_x = 4095
set accgain_y = 4100
set accgain_z = 4099
set align_mag = CW270FLIP
set mag_hardware = QMC5883
set magzero_x = -296
set magzero_y = -316
set magzero_z = -10
set maggain_x = 1289
set maggain_y = 1392
set maggain_z = 1360
set align_mag_roll = 250
set align_mag_pitch = 1800
set align_mag_yaw = 900
set baro_hardware = DPS310
set receiver_type = NONE
set serialrx_provider = CRSF
set blackbox_rate_denom = 2
set motor_pwm_protocol = DSHOT600
set failsafe_procedure = RTH
set align_board_yaw = 1800
set applied_defaults = 5
set gps_sbas_mode = AUTO
set airmode_type = THROTTLE_THRESHOLD
set osd_units = IMPERIAL
set name = REKON10
set pilot_name = BSOKFPV

# mixer_profile
mixer_profile 1

set model_preview_type = 3

# Mixer: motor mixer

mmix reset

mmix 0  1.000 -1.000  1.000 -1.000
mmix 1  1.000  1.000  1.000  1.000
mmix 2  1.000 -1.000 -1.000  1.000
mmix 3  1.000  1.000 -1.000 -1.000

# Mixer: servo mixer

# mixer_profile
mixer_profile 2


# Mixer: motor mixer

# Mixer: servo mixer

# profile
profile 1

set mc_p_pitch = 44
set mc_i_pitch = 85
set mc_d_pitch = 28
set mc_i_roll = 75
set mc_d_roll = 26
set mc_p_yaw = 40
set mc_i_yaw = 80
set dterm_lpf_hz = 80
set dterm_lpf_type = PT3
set mc_iterm_relax = RPY
set d_boost_min =  0.800
set d_boost_max =  1.200
set antigravity_gain =  2.000
set antigravity_accelerator =  5.000
set smith_predictor_delay =  1.500
set tpa_rate = 20
set tpa_breakpoint = 1200
set rc_expo = 75
set rc_yaw_expo = 75
set roll_rate = 70
set pitch_rate = 70
set yaw_rate = 60

# profile
profile 2


# profile
profile 3


# battery_profile
battery_profile 1

set battery_capacity = 8400
set battery_capacity_warning = 5880
set battery_capacity_critical = 6720

# battery_profile
battery_profile 2


# battery_profile
battery_profile 3


# restore original profile selection
mixer_profile 1
profile 1
battery_profile 1

# save configuration
save

**Reply by MrD-RC** [reply]
The OSD chip on the flight controller has nothing to do with HD video systems. The OSD is sent to the VTX over the UART. It is then sent to the goggles over the air. The goggles then adds the OSD  elements over the image.

**Reply by Blake1532** [reply]
Hello,

Oh, okay, so you are saying that the OSD chip only affects analog systems?  In that case should I make sure that the flight controller isn't blocking rx and tx for some reason?  And if it is, how would I check?

Thanks,
Blake

**Comment by Jetrell** [comment]
> I then uploaded betaflight, and I can't seem to get it working on betaflight either.

If you can't get it to work with Betaflight now, seems to indicate the problem is with the font package or setup in the goggles.

Are you using Avatar goggle firmware within the last two release ?  If so, this font package below, has to go onto the Goggle SD card.
Then you will have to use the menu in the goggle to load the font of your choice, from that font package... Whether it be Betaflight, INAV or Ardupilot, depending on what flight software your model is using.


[userfont.zip](https://github.com/iNavFlight/inav/files/14043841/userfont.zip)

**Comment by Blake1532** [comment]
Hello,

I am on the latest release as I was told to by caddx to fix the overheating
issue in the Goggles X.  I'll go ahead and install that font package on my
sd card and let you know if it works!

Thank you so much!
Blake

**Reply by Blake1532** [reply]
Hello,

Welll, that didn't work.  I don't know why this is happening.  It could be the vtx I suppose?

Thanks,
Blake

**Comment by fruts88** [comment]
Hello.
I have 3 FC Mamaba MK4 F722 MINI, WING and G4 Alpha.
Does not work OSD on Avatar with FC INAV7 (6, 6.1) and FETTEC.
With BTFL OSD works. 
The problem is in the firmware BTFL and INAV of the flight controller.
In BTFL pin VTX FC is the same as pin RX3 FC, but in the firmware INAV pin VTX and pin RX3 are different.
I soldered one wire to RX3 FC and the osd INAV started working.
 
Do you know which command are doing it in the CLI line INAV? 
I’am afraid I can’t solder the wires on other units (I haven't pin connector).
 
Than you.

**Reply by MrD-RC** [reply]
There are no CLI commands needed for walksnail on INAV. The issue is you’re trying to set it up using BetaFlight instructions. Follow this guide https://youtu.be/elq6AF5gdXE?si=sPLY-O7mw-X4Ypph

**Comment by pampas78** [comment]
Hi. I have the same issue with nav 7.1 and speedy bee wing mini. There is no way to see OSD.

**Reply by sensei-hacker** [reply]
Please see the video Darren linked to.
In particular, if you're accustomed to Betaflight, do NOT set that UART and an MSP control link.

**Comment by pampas78** [comment]
![Sin título](https://github.com/iNavFlight/inav/assets/129760146/743d0d1c-c46d-4f33-94f5-5eb75c7312e1)
It is solved!. I´ve connected first on the 4 pin (analog vtx) and changed to the 6 pin connector.

**Reply by QuocSiPHU** [reply]
Could you give us more details. Thanks

**Reply by pampas78** [reply]
[IMG_5776 (1).mov.zip](https://github.com/iNavFlight/inav/files/15370529/IMG_5776.1.mov.zip)
On the 6 pin connector use Vv-GND-TX5-RX5, cut the other 2 wires. Solder to the 4pin Caddx wriest corresponding 6pin and you are done.

**Reply by Dumbtoom** [reply]
Hello,
in case this topic is still of interest. I was having the same frustrating experience and when I read the post above my first thought was: OK, what pampas 78 describes is exactly what is mentioned in the manuals of speedy bee and walksnail. Then I realized that I had chosen to power the walksnail transmitte directly from the flight battery. 
I had only Tx and RX connected to the flight controller (speedy bee wing mini).
Then I tested is using the power supply from the speedy bee UART5 (Pin Vv and GND).
And it worked.
I have not tested it but I am assuming the GND connection between Speedy Bee and Walksnail Transmitter is important.

After hours of desperately changing settings in INAV, trying older versions etc. I think this is one of thoses cases when it works as soon as you do it right (as the f.. manual says ) :-)

**Reply by gyzmo82** [reply]
Hi All, I went through the above steps with my AtomRC Penguin and Moonlight VTX and no luck. Connected the VTX to UART5 , downloaded the fonts from the above section. Can you please help me if there are any other steps to follow?

**Reply by MrD-RC** [reply]
@gyzmo82 as yours is a completely different issue. It would be better to start a fresh discussion with all your details, fc, what you tried, etc. 

Setting up Walksnail with INAV should be pretty straightforward. https://youtu.be/elq6AF5gdXE?si=oDhTvg9laFMwaoz0

**Reply by RobKita** [reply]
My experience is a bit funny..
It turned out that I had the osd turned off in the iNav "modes" config,, it was an old setting for my analog vtx.
Since it was way down in the list it took me long time to discover.
